### PR TITLE
Ww setup linter

### PR DESCRIPTION
**Pull request by wwhorton**
Adds the latest version of ESLint and Prettier, and corrects subsequent errors and warnings.

## Additions

- ESLint
- Prettier

## Changes

- Modifies existing code to meet ESLint rules, esp. variable name length and unnamed default exports
- Disables rules where implementation would be counterproductive or unnecessary
- Fleshes out JSDOC statements

## Testing

- ```yarn test``` should pass
- ```yarn lint``` should pass
- Browse through app and check that it works and looks the same. Please pay particular attention to icons and buttons/anchors as some refactoring was necessary that might have affected styling. 

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/components/Map/MapToolbar.js [hunk 1]
```diff
@@ -21,28 +21,28 @@ export const MapToolbar = () => {
       <section className="state-heading">
         {!filteredStates && <span>United States of America</span>}
         <span>{filteredStates}</span>
-        {filteredStates && (
-          <a
+        {!!filteredStates && (
+          <button
```
Comment (flacoman91) [inline src/components/Map/MapToolbar.js:25]: This breaks the styling

Screenshot 2024-02-07 at 3 19 25 PM


It should look like
Screenshot 2024-02-07 at 3 20 19 PM

Go to the map tab and click some state tiles to see where the links should appear.

**Review by flacoman91 — CHANGES_REQUESTED** [review]

File: src/components/Charts/LineChart.js [hunk 2]
```diff
@@ -177,6 +179,7 @@ export const mapStateToProps = (state) => {
   };
 };
 
+// eslint-disable-next-line react-redux/prefer-separate-component-file
```
[the remark this answers is not in the document]
Comment (wwhorton) [reply src/components/Charts/LineChart.js:182]: Yeah, that would be a good project. I'd think we could move a lot of the Explorer functional components into CCDB pretty easily.

File: src/components/Map/MapToolbar.js (hunk 1 above)
Comment (wwhorton) [reply src/components/Map/MapToolbar.js:25]: Ok, these are fixed.

**Review by flacoman91 — APPROVED** [review]
👍   Looks good to me

**Pull request merged**